packages/client/src/build-id-types.ts

```diff
@@ -0,0 +1,137 @@
+import { temporal } from '@temporalio/proto';
+
+/**
+ * Operations that can be passed to {@link WorkflowClient.updateWorkerBuildIdCompatability}.
+ *
+ * @experimental
+ */
+export type BuildIdOperation =
+  | NewIdInNewDefaultSet
+  | NewCompatibleVersion
+  | PromoteSetByBuildId
+  | PromoteBuildIdWithinSet
+  | MergeSets;
+
+/**
+ * This operation adds a new Build Id into a new set, which will be used as the default set for
+ * the queue. This means all new workflows will start on this Build Id.
+ *
+ * @experimental
+ */
+interface NewIdInNewDefaultSet {
+  operation: 'NEW_ID_IN_NEW_DEFAULT_SET';
+  buildId: string;
+}
+
+/**
+ * This operation adds a new Build Id into an existing compatible set. The newly added ID becomes
+ * the default for that compatible set, and thus new workflow tasks for workflows which have been
+ * executing on workers in that set will now start on this new Build ID.
```

Comment:
Let's consistently use `Id` please.
```suggestion
 * executing on workers in that set will now start on this new Build Id.
```